src/test/java/kitchenpos/application/menu/MenuQueryServiceTest.java

```diff
@@ -0,0 +1,71 @@
+package kitchenpos.application.menu;
+
+import kitchenpos.application.MenuService;
+import kitchenpos.config.ApplicationTestConfig;
+import kitchenpos.domain.Menu;
+import kitchenpos.domain.MenuGroup;
+import kitchenpos.domain.MenuProduct;
+import kitchenpos.domain.Product;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.assertj.core.api.SoftAssertions.assertSoftly;
+
+class MenuQueryServiceTest extends ApplicationTestConfig {
+
+    private MenuService menuService;
+
+    @BeforeEach
+    void setUp() {
+        menuService = new MenuService(
+                menuDao,
+                menuGroupDao,
+                menuProductDao,
+                productDao
+        );
+    }
```

Comment:
ApplicationTestConfig 가 @SpringBootTest 를 사용하기 때문에 setUp 없이 바로 의존성 주입이 가능합니다!!😊

```suggestion
    @Autowired
    private MenuService menuService;
```

Reply:
서비스에서 `@Autowired`를 선호하는 편이 아니라서 사용하지 않았습니다. 🙇‍♂️
물론 테스트 코드이고 `@Autowired`의 편의성은 엄청나다고 생각합니다.

하지만 이런식으로 dao를 직접 주입하여 Service를 생성할 경우 
해당 서비스에 어떠한 데이터들이 필요하게 될지 테스트 코드만 봐도 예상이 가능합니다.

또한 JPA를 사용한다고 가정했을 때 서비스 테스트에서도 `@DataJpaTest`를 이용하여 테스트 비용을 절감시킬 수 있습니다.

때문에 저는 이와 같은 방식으로 진행하는 편이 경험상 제일 좋았던거 같습니다.

Reply:
두가지 모두 좋은 장점이네요!! 헤나의 의견 공유해주셔서 감사합니다😊

Mocking 테스트 또한 헤나가 말해주신 두가지 장점 모두 살릴 수 있는 테스트 처럼 보이는데 이번 미션에서 Mocking 을 사용하지 않은 이유도 궁금해요!!😊

Reply:
스플릿 말씀을 들으니 Mocking 테스트를 왜 안쓰게 됐는지 생각하게 됐네요! 👍

저는 도메인과 엔티티를 구분하기 위해서 객체를 생성할 때  식별자값을 넣지 않게 했습니다.
도메인은 직접 생성하고, 엔티티는 데이터베이스 접근 계층에서 만들어지게하여 실수 가능성을 낮췄습니다.

Mocking을 하면 SpringBootTest 보다 테스트 실행 속도는 빠르겠지만 번거로움이 증가하는거 같습니다.
DAO를 Mocking하여 엔티티를 반환했을 때 식별자값을 가져오는 로직이 존재한다면 식별자를 가져오는 getter 메서드까지 고려해야 합니다.
만약 SpringBootTest로 실제 데이터베이스를 이용해서 테스트를 진행한다면 단순히 데이터베이스에서 가져와서 사용하기만 하면 됩니다.

또한 Mocking은 서비스 내부 로직을 테스트에 노출시키고 있어 서비스 기능 리팩터링할 경우 테스트 코드까지 변경될 가능성이 큽니다.

이렇게 Mocking은 테스트 실행 속도는 훨씬 빠르지만 코드 생산성이나 유연성은 떨어진다고 생각합니다.
그렇다고해서 실제 객체를 사용하는 것이 무조건 좋다고 생각하지는 않습니다.
극단적으로 생각해봤을 때 SpringBootTest에서 서비스를 테스트할 때 데이터를 세팅하려고 하는데 간단한 테스트임에도 외래키 조건이 100개 정도라고 하면 그에 대한 값을 모두 세팅하는 것보다 Mocking을 사용하는 편이 좋은거 같습니다.